@types/index.d.ts

```diff
@@ -131,52 +139,37 @@ interface Request extends Body {
 	 * Returns the URL of request as a string.
 	 */
 	readonly url: string;
-	clone: () => Request;
+	clone(): Request;
 }
-type RequestInfo = string | Body;
-declare var Request: {
-	prototype: Request;
-	new (input: RequestInfo, init?: RequestInit): Request;
-};
 
-interface Response extends Body {
+export class Response extends Body {
+	constructor(body?: BodyInit | null, init?: ResponseInit);
+
 	readonly headers: Headers;
 	readonly ok: boolean;
 	readonly redirected: boolean;
 	readonly status: number;
 	readonly statusText: string;
 	readonly url: string;
-	clone: () => Response;
-}
-
-declare var Response: {
-	prototype: Response;
-	new (body?: BodyInit | null, init?: ResponseInit): Response;
-};
-
-declare function fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
-
-declare namespace fetch {
-	function isRedirect(code: number): boolean;
+	clone(): Response;
 }
 
-interface FetchError extends Error {
+export class FetchError extends Error {
+	constructor(message: string, type: string, systemError?: object);
+	
 	name: 'FetchError';
 	[Symbol.toStringTag]: 'FetchError';
 	type: string;
 	code?: string;
 	errno?: string;
 }
-declare var FetchError: {
-	prototype: FetchError;
-	new (message: string, type: string, systemError?: object): FetchError;
-};
 
 export class AbortError extends Error {
 	type: string;
 	name: 'AbortError';
 	[Symbol.toStringTag]: 'AbortError';
 }
 
-export {Headers, Request, Response, FetchError};
-export default fetch;
+export function isRedirect(code: number): boolean;
+
+export default function fetch(url: RequestInfo, init?: RequestInit): Promise<Response>;
```

Comment:
```suggestion
declare type Fetch = (url: RequestInfo, init?: RequestInit) => Promise<Response>;
declare const fetch: Fetch & { default: Fetch };
export = fetch;
```

Reply:
I found another way that allows to export types along with `fetch`. See my next comment.